### PR TITLE
Implement weekranges for GGD data

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

This PR implements the latest schema changes for GGD data. 

## Motivation

Feature request

## Detailed design

The line charts were already showing week dates, but these were previously calculated on the fly. Now that the data properly contains week_start and week_end values the line charts have been re-refactored to using these.
lineChartWithWeekTooltip now also lets its tooltip formatting function to optionally be prop injected, this in order to show percentages.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: src/components/lineChart/multipleLineChart.tsx [hunk 1]
```diff
@@ -90,21 +84,26 @@ function getChartOptions(
...
       useHTML: true,
       formatter: function (): string {
-        const percentage =
-          ((this.points as any[])[1].y * 100) / (this.points as any[])[0].y;
+        const contextObjects = this.points as TooltipFormatterContextObject[];
+
+        const percentage = (contextObjects[1].y * 100) / contextObjects[0].y;
+
+        const { originalData }: { originalData: Value } = contextObjects[0]
+          .point as any;
```
Comment (VWSCoronaDashboard4) [inline src/components/lineChart/multipleLineChart.tsx:92]:
```suggestion
          .point as { originalData: Value };
```

File: src/pages/veiligheidsregio/[code]/positief-geteste-mensen.tsx [hunk 2]
```diff
@@ -204,11 +203,23 @@ const PostivelyTestedPeople: FCWithLayout<ISafetyRegionData> = (props) => {
...
+            value: value.infected_percentage,
+            date: value.week_unix,
+            week: {
+              start: value.week_start_unix,
+              end: value.week_end_unix,
+            },
           }))}
-          formatTooltip={(x: number, y: number) => {
-            return `${formatDateFromSeconds(x)}: ${formatPercentage(y)}%`;
+          tooltipFormatter={function () {
```
Comment (VWSCoronaDashboard4) [inline src/pages/veiligheidsregio/[code]/positief-geteste-mensen.tsx:213]:
```suggestion
          tooltipFormatter={() => {
```

Comment (VWSCoronaDashboard1) [reply src/pages/veiligheidsregio/[code]/positief-geteste-mensen.tsx:213]: Nope, the formatter function gets its context set to the 'this' instance, you can't set 'this' on a lambda

**Review by VWSCoronaDashboard8 — APPROVED** [review]

File: src/pages/landelijk/positief-geteste-mensen.tsx [hunk 3]
```diff
@@ -246,11 +249,23 @@ const PositivelyTestedPeople: FCWithLayout<INationalData> = (props) => {
...
+            date: value.week_unix,
+            week: {
+              start: value.week_start_unix,
+              end: value.week_end_unix,
+            },
           }))}
-          formatTooltip={(x: number, y: number) => {
-            return `${formatDateFromSeconds(x)}: ${formatPercentage(y)}%`;
+          tooltipFormatter={function () {
+            const { originalData }: { originalData: Value } = this.point as any;
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply src/pages/landelijk/positief-geteste-mensen.tsx:260]: hehe, I had already changed it to exactly that a minute ago ;)

File: src/pages/veiligheidsregio/[code]/positief-geteste-mensen.tsx [hunk 4]
```diff
@@ -204,11 +203,23 @@ const PostivelyTestedPeople: FCWithLayout<ISafetyRegionData> = (props) => {
...
+            const { originalData } = this.point as any;
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard1) [reply src/pages/veiligheidsregio/[code]/positief-geteste-mensen.tsx]: Also fixed

**Pull request merged**